.github/workflows/vast.yaml

```diff
@@ -93,73 +103,139 @@ jobs:
           done
-          echo "::set-output name=docker-build-args::${docker_build_args}"
-          echo "::set-output name=vast-container-ref::${vast_container_ref}"
+          echo "cmake-version-args=${cmake_version_args}" >> $GITHUB_OUTPUT
+          docker_build_args="${cmake_version_args} -D VAST_ENABLE_AVX_INSTRUCTIONS:BOOL=OFF"
+          docker_build_args="${docker_build_args} -D VAST_ENABLE_AVX2_INSTRUCTIONS:BOOL=OFF"
+          echo "docker-build-args=${docker_build_args}" >> $GITHUB_OUTPUT
+          echo "vast-container-ref=${vast_container_ref}" >> $GITHUB_OUTPUT
+          # Decide what jobs to run
+          echo "run-python=true" >> $GITHUB_OUTPUT
+          run_python_package=false
+          if [[ $GITHUB_EVENT_NAME == "release" ]]; then
+            run_python_package=true
+          fi
+          run_example_notebooks=true
+          if [[ $GITHUB_EVENT_NAME == "workflow_dispatch" ]]; then
+            run_example_notebooks=false
+          echo "run-example-notebooks=${run_example_notebooks}" >> $GITHUB_OUTPUT
+          echo "run-python-package=${run_python_package}" >> $GITHUB_OUTPUT
+          if [[ "$GITHUB_EVENT_NAME" != "pull_request" ]]; then
+            echo "run-changelog=true" >> $GITHUB_OUTPUT
+            echo "run-vast-io=true" >> $GITHUB_OUTPUT
+            echo "run-docker=true" >> $GITHUB_OUTPUT
+            echo "run-docker-compose=true" >> $GITHUB_OUTPUT
+            echo "run-vast-nix=true" >> $GITHUB_OUTPUT
+            echo "run-vast=true" >> $GITHUB_OUTPUT
+            echo "run-vast-plugins=true" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          shopt -s expand_aliases
+          alias is_unchanged="git diff --exit-code ${before_sha} --"
+          run_vast_io=false
+          if ! is_unchanged web/; then
+            run_vast_io=true
+          fi
+          echo "run-vast-io=${run_vast_io}" >> $GITHUB_OUTPUT
+          run_changelog=${run_vast_io}
+          if ! is_unchanged CHANGELOG.md; then
+            run_changelog=true
+          elif ! is_unchanged changelog/; then
+            run_changelog=true
+          fi
+          echo "run-changelog=${run_changelog}" >> $GITHUB_OUTPUT
+          run_vast=false
+          if ! is_unchanged cmake/; then
+            run_vast=true
+          elif ! is_unchanged CMakeLists.txt; then
+            run_vast=true
+          elif ! is_unchanged libvast/; then
+            run_vast=true
+          elif ! is_unchanged libvast_test/; then
+            run_vast=true
+          elif ! is_unchanged scripts/; then
+            run_vast=true
+          elif ! is_unchanged schema/; then
+            run_vast=true
+          elif ! is_unchanged tools/; then
+            run_vast=true
+          elif ! is_unchanged vast/; then
+            run_vast=true
+          elif ! is_unchanged vast.yaml.example; then
+            run_vast=true
+          elif ! is_unchanged version.json; then
+            run_vast=true
+          fi
```

Comment:
For all instances of is_unchanged here, do you think it'd make sense to group them together, i.e., `if ! is_unchanged cmake/ CMakeLists.txt libvast/ ...`?

Reply:
Yes, i think this is better without this many if elifs :laughing: